src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -122,51 +103,21 @@ private Task ProcessRequestAsync(StreamingMessage request)
 
         private async Task ProcessRequestCoreAsync(StreamingMessage request)
         {
-            StreamingMessage responseMessage = new StreamingMessage
-            {
-                RequestId = request.RequestId
-            };
-
-            if (request.ContentCase == MsgType.InvocationRequest)
-            {
-                responseMessage.InvocationResponse = await InvocationRequestHandlerAsync(request.InvocationRequest, _application,
-                    _invocationFeaturesFactory, _serializer, _outputBindingsInfoProvider, _inputConversionFeatureProvider);
-            }
-            else if (request.ContentCase == MsgType.WorkerInitRequest)
+            var handler = _factory.CreateHandler(request.ContentCase);
+            if (handler == null)
             {
-                responseMessage.WorkerInitResponse = WorkerInitRequestHandler(request.WorkerInitRequest);
-            }
-            else if (request.ContentCase == MsgType.WorkerStatusRequest)
-            {
-                responseMessage.WorkerStatusResponse = new WorkerStatusResponse();
-            }
-            else if (request.ContentCase == MsgType.FunctionsMetadataRequest)
-            {
-                responseMessage.FunctionMetadataResponse = await GetFunctionMetadataAsync(request.FunctionsMetadataRequest.FunctionAppDirectory);
-            }
-            else if (request.ContentCase == MsgType.WorkerTerminate)
-            {
-                WorkerTerminateRequestHandler(request.WorkerTerminate);
-            }
-            else if (request.ContentCase == MsgType.FunctionLoadRequest)
-            {
-                responseMessage.FunctionLoadResponse = FunctionLoadRequestHandler(request.FunctionLoadRequest, _application, _methodInfoLocator);
-            }
-            else if (request.ContentCase == MsgType.FunctionEnvironmentReloadRequest)
-            {
-                // No-op for now, but return a response.
-                responseMessage.FunctionEnvironmentReloadResponse = new FunctionEnvironmentReloadResponse
-                {
-                    Result = new StatusResult { Status = StatusResult.Types.Status.Success }
-                };
+                // TODO: Trace failure here.
+                return;
             }
-            else
+
+            var response = await handler.HandleMessageAsync(request);
+            if (response == null)
```

Comment:
Can the response be  null ? If so, we should consider updating `IGrpcWorkerMessageHandler.HandleMessageAsync` return type to reflect that.

Reply:
It seems like it can't. The current logic has no checks for null/does not return null response. Will remove the check.